web/src/user_profile.ts

```diff
@@ -99,15 +129,28 @@ export function update_user_profile_streams_list_for_users(user_ids) {
     }
 }
 
-export function update_profile_modal_ui(user, new_data) {
+export function update_profile_modal_ui(
+    user: User,
+    new_data: {
+        user_id?: number;
+        bot_owner_id?: string;
+        avatar_url?: string;
+        delivery_email?: string;
+        role?: number;
+        full_name?: string;
+    },
+): void {
     if (!(modals.any_active() && modals.active_modal() === "#user-profile-modal")) {
         return;
     }
     if (original_values?.user_id === undefined) {
         // This occurs if say, the "channel" tab is open.
         return;
     }
-    const current_user_id = Number.parseInt(original_values.user_id, 10);
+    const current_user_id =
+        typeof original_values.user_id === "number"
+            ? original_values.user_id
+            : Number.parseInt(original_values.user_id, 10);
```

Comment:
This seems like it could be cleaned up -- we should be able to know whether `original_values` has an integer or a string-format integer, right?

Reply:
I tried testing manually and the `user_id` seems to be `string` always, I've changed the type to `string`.